dot/rpc/modules/grandpa.go

```diff
@@ -48,39 +48,32 @@ type RoundStateResponse struct {
 
 // ProveFinalityRequest request struct
 type ProveFinalityRequest struct {
-	blockHashStart common.Hash
-	blockHashEnd   common.Hash
-	authorityID    uint64
+	BlockNumber uint32 `json:"blockNumber"`
 }
 
 // ProveFinalityResponse is an optional SCALE encoded proof array
-type ProveFinalityResponse [][]byte
+type ProveFinalityResponse []string
 
-// ProveFinality for the provided block range. Returns NULL if there are no known finalised blocks in the range.
-// If no authorities set is provided, the current one will be attempted.
+// ProveFinality for the provided block number, returning the Justification for the last block in the set.
```

Comment:
nit please clarify this a bit, no justification is 'returned' from this function, I guess it's written to the response. Also that response can contain a string error in case of failure.